polaris/polaris/utils.py

```diff
@@ -420,18 +445,68 @@ def create_transaction_envelope(transaction, source_account) -> TransactionEnvel
         source_account=source_account,
         network_passphrase=settings.STELLAR_NETWORK_PASSPHRASE,
         base_fee=base_fee,
-    ).append_payment_op(
-        destination=transaction.stellar_account,
-        asset_code=transaction.asset.code,
-        asset_issuer=transaction.asset.issuer,
-        amount=str(payment_amount),
-        source=transaction.asset.distribution_account,
     )
+    if (
+        transaction.claimable_balance_supported
+        and transaction.status == Transaction.STATUS.pending_trust
+    ):
+        transaction.status = Transaction.STATUS.pending_anchor
+        transaction.save()
+        logger.info(
+            f"Transaction {transaction.id} is {transaction.status}, crafting claimable balance operation"
+        )
+        claimant = Claimant(destination=transaction.stellar_account)
+        asset = Asset(code=transaction.asset.code, issuer=transaction.asset.issuer)
+        builder.append_create_claimable_balance_op(
+            claimants=[claimant],
+            asset=asset,
+            amount=str(payment_amount),
+            source=transaction.asset.distribution_account,
+        )
+    else:
+        builder.append_payment_op(
+            destination=transaction.stellar_account,
+            asset_code=transaction.asset.code,
+            asset_issuer=transaction.asset.issuer,
+            amount=str(payment_amount),
+            source=transaction.asset.distribution_account,
+        )
     if memo:
         builder.add_memo(memo)
     return builder.build()
 
 
+def get_balance_id(response):
+    """
+    Pulls claimable balance ID from horizon responses.
+
+    When called we decode and read the result_xdr from the horizon response.
+    If any of the operations is a createClaimableBalanceResult we
+    decode the Base64 representation of the balanceID xdr.
+    After the fact we encode the result to hex.
+
+    The hex representation of the balanceID is important because its the
+    representation required to query and claim claimableBalances.
+
+    :param
+        response: the response from horizon
+
+    :return:
+        hex representation of the balanceID
+        or
+        None (if no createClaimableBalanceResult operation is found)
```

Comment:
Thanks for adding documentation on parameter and return types, but I think the best way to do this is using Python's `typing` module. I added a suggestion to show you what I mean. The docstring above `:param` sufficiently explains whats happening, no `:param` or `:return:` necessary.